dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/bootstrap/BootstrapTakeoverState.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.config.bootstrap;
+
+import org.apache.dubbo.config.ServiceConfig;
+
+/**
+ * State of which of DubboBootstrap lifecycle being takeover
+ * SPRING: will be controlled by spring context
+ * MANUAL: will be controlled by users, after all services init, should call {@link DubboBootstrap#start()} to init app-level env
+ * AUTO: env will be init once {@link ServiceConfig#export()} finished
+ * SERVLET: will be controlled by tomcat
+ */
+public enum BootstrapTakeoverState {
```

Comment:
It may be more appropriate to modify to `BootstrapTakeoverMode`, not a state, but a mode.